### PR TITLE
vgmstream r1702

### PR DESCRIPTION
**Pull request by xxyzz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`autoconf` and `automake` are required by `vgmstream`'s dependency `celt`. Closes #94247

### CODE REVIEW

File: Formula/vgmstream.rb [hunk 1]
```diff
@@ -25,6 +37,8 @@ class Vgmstream < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "cb21895ea0256c867a24bc300cc5bc254c2e769e4e74a4cb2766ae9f5662fe57"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
```
Comment (carlocab) [inline Formula/vgmstream.rb:41]: Let's add a comment here that these can be removed when the patches are no longer needed.

File: Formula/vgmstream.rb [hunk 2]
```diff
@@ -35,11 +49,15 @@ class Vgmstream < Formula
...
-    lib.install "src/libvgmstream.a"
+    ENV["LIBRARY_PATH"] = "#{HOMEBREW_PREFIX}/lib"
+    Dir.mkdir("build")
+    Dir.chdir("build") do
+      system "cmake", "-DBUILD_AUDACIOUS:BOOL=OFF", ".."
+      system "make"
+      bin.install "cli/vgmstream-cli"
+      bin.install "cli/vgmstream123"
+      lib.install "src/libvgmstream.a"
+    end
```
Comment (carlocab) [inline Formula/vgmstream.rb:60]: Do we need to change this?

File: Formula/vgmstream.rb (hunk 1 above)
Comment (xxyzz) [reply Formula/vgmstream.rb:41]: They are needed to build `celt`, one of the dependencies of vgmstream, they can't be removed. The patches are for fixing errors caused by cmake, more details: https://github.com/vgmstream/vgmstream/issues/1069.

File: Formula/vgmstream.rb (hunk 2 above)
Comment (xxyzz) [reply Formula/vgmstream.rb:60]: vgmstream's [build document](https://github.com/vgmstream/vgmstream/blob/master/doc/BUILD.md#macos) suggests these command and [doesn't recommend build in root folder](https://github.com/vgmstream/vgmstream/blob/master/doc/BUILD.md#cmake-builds).

Comment (carlocab) [reply Formula/vgmstream.rb:60]: Ok, let's do this instead then
```suggestion
    system "cmake", "-S", ".", "-B", "build", "-DBUILD_AUDACIOUS:BOOL=OFF", *std_cmake_args
    system "cmake", "--build", "build"
    bin.install "build/cli/vgmstream-cli", "build/cli/vgmstream123"
    lib.install "build/src/libvgmstream.a"
```

File: Formula/vgmstream.rb [hunk 3]
```diff
@@ -35,11 +49,15 @@ class Vgmstream < Formula
   fails_with gcc: "5" # ffmpeg is compiled with GCC
 
   def install
-    system "cmake", "-DBUILD_AUDACIOUS:BOOL=OFF", *std_cmake_args, "."
-    system "cmake", "--build", ".", "--config", "Release", "--target", "vgmstream_cli", "vgmstream123"
-    bin.install "cli/vgmstream-cli"
-    bin.install "cli/vgmstream123"
-    lib.install "src/libvgmstream.a"
+    ENV["LIBRARY_PATH"] = "#{HOMEBREW_PREFIX}/lib"
```
Comment (carlocab) [inline Formula/vgmstream.rb:52]:
```suggestion
    ENV["LIBRARY_PATH"] = HOMEBREW_PREFIX/"lib"
```
Why do we need to set this?

Comment (xxyzz) [reply Formula/vgmstream.rb:52]: vgmstream's [maintainer's advice](https://github.com/vgmstream/vgmstream/issues/1069#issuecomment-1030501736), I guess it's for compiler to find library installed from Homebrew. Without this variable, it won't compile.

File: Formula/vgmstream.rb [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
+
+    # patches for fixing macOS build error, remove them at next release
+    patch do
+      url "https://github.com/vgmstream/vgmstream/commit/04b5a1f098f348c4c438ada85ee86f5a9abce2ff.patch?full_index=1"
+      sha256 "37d5d9567435eda96bf4e51b5abbac1677391e08daf4e50b1d7edc6996772919"
+    end
+    patch do
+      url "https://github.com/vgmstream/vgmstream/commit/f9230158d953dfb0fb14a2a57052a8460a5d43dd.patch?full_index=1"
+      sha256 "10ff07d7c6f48275af235d720f279ef62d444c88cde78f2fecaf065e120d134d"
+    end
```
Comment (carlocab) [inline Formula/vgmstream.rb:22]: We actually set `CC` and `CXX` in the environment, so this patch probably doesn't do anything. It doesn't hurt, though, so there's no need to remove it.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**